src/pages/AccountDashboard/DailyAndCumulativePnL.tsx

```diff
@@ -0,0 +1,303 @@
+      <StatsTooltipRow label={t`Date`} value={stats.date} showDollar={false} />
+      <StatsTooltipRow
+        label={t`PnL`}
+        value={formatUsd(stats.pnl)}
+        showDollar={false}
+        textClassName={getPositiveOrNegativeClass(stats.pnl)}
+      />
+      <StatsTooltipRow
+        label={t`Cumulative PnL`}
+        value={formatUsd(stats.cumulativePnl)}
+        showDollar={false}
+        textClassName={getPositiveOrNegativeClass(stats.cumulativePnl)}
+      />
+      <DebugTooltip stats={stats} />
+    </div>
+  );
+}
+
+export type AccountPnlHistoryPoint = {
+  date: string;
+  dateCompact: string;
+  pnlFloat: number;
+  pnl: bigint;
+  cumulativePnlFloat: number;
+  cumulativePnl: bigint;
+} & AccountPnlHistoryPointDebugFields;
+
+type PnlHistoricalData = FromOldToNewArray<AccountPnlHistoryPoint>;
+
+const PROD_QUERY = gql`
+  query AccountHistoricalPnlResolver($account: String!, $from: Int) {
+    accountPnlHistoryStats(account: $account, from: $from) {
+      cumulativePnl
+      pnl
+      timestamp
+    }
+  }
+`;
+
+function usePnlHistoricalData(chainId: number, account: Address, fromTimestamp: number | undefined) {
+  const showDebugValues = useShowDebugValues();
+  const res = useGqlQuery(showDebugValues ? DEV_QUERY : PROD_QUERY, {
+    client: getSubsquidGraphClient(chainId)!,
+    variables: { account: account, from: fromTimestamp },
+  });
+
+  const transformedData: PnlHistoricalData = useMemo(() => {
+    return (
+      res.data?.accountPnlHistoryStats?.map((row: any) => {
+        const parsedDebugFields = showDebugValues
+          ? DEBUG_FIELDS.reduce(
+              (acc, key) => {
+                const raw = row[key];
+
+                const bn = raw ? BigInt(raw) : 0n;
+                acc[key] = bn;
+                acc[`${key}Float`] = usdBigIntToFloat(bn);
+                return acc;
+              },
+              {} as Record<string, bigint | number>
+            )
+          : EMPTY_OBJECT;
+
+        return {
+          date: showDebugValues
+            ? formatDateTime(row.timestamp) + " - " + formatDateTime(row.timestamp + 86400) + " local"
+            : formatDate(row.timestamp),
+          dateCompact: lightFormat(row.timestamp * 1000, "dd/MM"),
+          pnl: BigInt(row.pnl),
+          pnlFloat: usdBigIntToFloat(BigInt(row.pnl)),
+          cumulativePnl: BigInt(row.cumulativePnl),
+          cumulativePnlFloat: usdBigIntToFloat(BigInt(row.cumulativePnl)),
+          ...parsedDebugFields,
+        };
+      }) || EMPTY_ARRAY
+    );
+  }, [res.data?.accountPnlHistoryStats, showDebugValues]);
+
+  return { data: transformedData, error: res.error, loading: res.loading };
+}
+
+function usdBigIntToFloat(usd: bigint) {
+  if (typeof usd !== "bigint") {
+    throw new Error(`usdBigIntToFloat: expected bigint, got ${typeof usd}, ${usd}`);
+  }
+  return Number((usd * 1_0000n) / 10n ** BigInt(USD_DECIMALS)) / 1_0000;
```

Comment:
we have now tested `bigintToNumber` 
https://github.com/gmx-io/gmx-interface/blob/master/src/lib/numbers.ts#L466

I think it's applicable here

Reply:
done